### ISSUE TITLE
Add exclude_files option to file integrity module

### ISSUE THREAD

**Issue by andrewkroh** [elastic/beats#5342]
An `exclude_files` configuration option should be added to the file integrity module in Auditbeat to ignore files that match any of the regular expressions specified in the list.

For example:

```
auditbeat.modules:
- module: audit
  metricsets: [file]
  file.paths:
  - /bin
  - /usr/bin
  - /usr/local/bin
  file.exclude_files: ['\.DS_Store$', '\.swp$']
```

Naming wise I'm debating between `file.exclude` and `file.exclude_files`. The latter is consistent with [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-filebeat-options.html#exclude-files).

**Comment by liketic** [comment]
Seem like `fsnotify` does not expose an exclude API. We can only do filter after received events for now.

**Comment by andrewkroh** [comment]
Right, for fsnotify we can filter the incoming events based on their `Name`. For scanning it will be similar. During the `filepath.Walk` we can skip anything that matches one of the regexes.

But the scanning feature (#5344) hasn't merged yet so I would hold off a second to avoid any merge conflicts (I did a lot of refactoring).

**Comment by adriansr** [comment]
I'm wondering about case-sensitivity for this regular expressions.
Can't decide what makes more sense. Ask the user to include (?i) at the start of every expression if case-insensitive matching is desired, or force case-insensitive matching in our side?

**Comment by andrewkroh** [comment]
For consistency with other parts of Beats I think the user should have to add `(?i)`. AFAIK this is how other areas work. [Looks](https://github.com/exekias/beats/blob/edde8912793f3055f08937fb88fe432e47a8baaa/filebeat/harvester/reader/multiline_config.go#L16) like we use `match.Matcher` in the config and go-ucfg will validate the expression.

**Comment by andrewkroh** [comment]
Implemented in #5888
